File: task2.js
const args = process.argv;

function fib(n) {
    if (Math.abs(n) < 2) return Math.abs(n);

    let a = 0;
    let b = 1;
    const nAbs = Math.abs(n)
    for (let i = 2; i <= nAbs; i++) {
        let c = a + b;
        a = b;
        b = c;
    }
    return n < 0 ? b*(Math.pow(-1, n + 1)) : b;
}

if (args[2]) process.stdout.write(fib(parseInt(args[2])).toString());

File: task2.rb
def fast_fib(fib_n)
    fib_n < 0 ? ((-1)**(fib_n + 1)) * _fib(fib_n.abs)[0] : _fib(fib_n)[0]
end
  
def _fib(n)
    if n == 0
        return [0, 1]
    else
        a, b = _fib(n / 2)
        c = a * (b * 2 - a)
        d = a * a + b * b

        if n % 2 == 0
            [c, d]
        else
            [d, c + d]
        end
    end
end

puts fast_fib(ARGV[0].to_i)